### PR TITLE
packages: switch all cmake packages to ninja

### PR DESCRIPTION
**Pull request by Andarwinux**
tested in a local build, Not sure how much it will speed up the GHA build.
libssh and some unused packages still use gnu make.
libarchive also switch to cmake+ninja.

### CODE REVIEW

**Comment by Andarwinux** [comment]
Fixed gcc build problem

**Comment by shinchiro** [comment]
Fix your pc date

**Comment by Andarwinux** [comment]
done.

**Comment by shinchiro** [comment]
Do you test with gcc? I got error in configuring ffmpeg with libzimg

**Comment by Andarwinux** [comment]
> Do you test with gcc? I got error in configuring ffmpeg with libzimg

Yes, have tested the x86-64 build with gcc and no problems. Since this PR does not make any changes to zimg and graphengine, this could be due to something else, can you check ffbuild/config.log?

File: packages/libarchive.cmake [hunk 1]
```diff
@@ -6,24 +6,39 @@ ExternalProject_Add(libarchive
...
-        --without-cng
-        --without-openssl
-    BUILD_COMMAND ${MAKE}
-    INSTALL_COMMAND ${MAKE} install
-    BUILD_IN_SOURCE 1
+    CONFIGURE_COMMAND ${EXEC} cmake -H<SOURCE_DIR> -B<BINARY_DIR>
+        -G Ninja
+        -DCMAKE_INSTALL_PREFIX=${MINGW_INSTALL_PREFIX}
+        -DCMAKE_FIND_ROOT_PATH=${CMAKE_INSTALL_PREFIX}
+        -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE}
```
Comment (shinchiro) [inline packages/libarchive.cmake:18]: Put `DCMAKE_BUILD_TYPE` here before project's cmake option for consistency. Apply to all cmake build

**Comment by Andarwinux** [comment]
tested with clang 17, clang 18, gcc 13

File: packages/xz.cmake [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+    CONFIGURE_COMMAND ${EXEC} cmake -H<SOURCE_DIR> -B<BINARY_DIR>
+        -G Ninja
+        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE}
+        -DCMAKE_INSTALL_PREFIX=${MINGW_INSTALL_PREFIX}
+        -DCMAKE_FIND_ROOT_PATH=${MINGW_INSTALL_PREFIX}
+        -DBUILD_SHARED_LIBS=OFF
+        -DBUILD_TESTING=OFF
+        -DENABLE_THREADS=ON
+        -DCMAKE_BUILD_TYPE=Release
```
Comment (shinchiro) [inline packages/xz.cmake]: duplicate

**Comment by shinchiro** [comment]
Got this error when configuring ffmpeg:
```
/home/anon/gcc_build64/install/x86_64-w64-mingw32/lib/libzimg.a(libzimg_internal_la-resize_impl.o): duplicate section `.rdata$_ZTS18graphengine_filter[_ZTS18graphengine_filter]' has different size
/home/anon/gcc_build64/install/x86_64-w64-mingw32/lib/libzimg.a(libzimg_internal_la-resize_impl.o): duplicate section `.rdata$_ZTSN11graphengine6FilterE[_ZTSN11graphengine6FilterE]' has different size
/home/anon/gcc_build64/install/x86_64-w64-mingw32/lib/libzimg.a(libzimg_internal_la-resize_impl.o): duplicate section `.rdata$_ZTIN11graphengine6FilterE[_ZTIN11graphengine6FilterE]' has different size
```
Not sure this come from my toolchain or not

**Comment by Andarwinux** [comment]
I can't reproduce the problem, maybe try clean build? This looks like a toolchain problem.
I am trying to reproduce the problem again using [Action](https://github.com/Andarwinux/test/actions/runs/6879425607/job/18711317609).

**Pull request closed without merging**